### PR TITLE
Data model day1

### PR DESCRIPTION
**Pull request by IsmailChangezi**
## Description 
- Setup and Intialise Rails
- Setup and Initialize Linters 
- Fix Linter Errors 
- Create Database 
- Create Migrations

# Queries 
### These are the queries that I have used:
- rails g migration CreateUsers name:string photo:string bio:text posts_counter:integer
- rails g migration CreatePosts title:string text:text comments_counter:integer likes_counter:integer
- rails g migration CreateComments text:text users:references posts:references
- rails g migration CreateLikes users:references posts:references

### CODE REVIEW

File: db/migrate/20230815113642_create_comments.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.references :users, null: false, foreign_key: true
```
Comment (muneebulrehman) [inline db/migrate/20230815113642_create_comments.rb]: As per the ER diagram provided to you, the relation between the comments and the user is set up through the author just like for the posts. Kindly update your code just like you did for the posts.

File: db/migrate/20230815113802_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :users, null: false, foreign_key: true
```
Comment (muneebulrehman) [inline db/migrate/20230815113802_create_likes.rb]: As per the ER diagram provided to you, the relation between the like and the user is set up through the author just like for the posts. Kindly update your code just like you did for the posts.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @IsmailChangezi , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Great job setting up the project
- Great job running the migration files.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @IsmailChangezi, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementingthechangesrequestedbythepreviousreviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Created migration file for `users`, `posts`, `comments` & `likes` ✔️
- Migrate all file to the schema ✔️ 
- Good use of Git flow ✔️

## Optional changes

- [ ] Please note, for future purposes kindly make your `commit` messages more descriptive and professional. here is a [link](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8) for more info

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**